lib/specinfra/backend/docker.rb

```diff
@@ -12,7 +12,9 @@ def initialize
       if image = Specinfra.configuration.docker_image
         @images = []
         @base_image = ::Docker::Image.get(image)
-        create_and_start_container
+
+        env = Specinfra.configuration.docker_env || {}
```

Comment:
>  I assumed that env was for the running context as opposed to inside the docker container.

No. `Specinfra.configuration.env` is for defining environment variables in test target machines.
So it is applicable to docker containers.